src/Symfony/Bundle/FrameworkBundle/EventListener/SessionListener.php

```diff
@@ -14,10 +14,14 @@
 use Symfony\Component\HttpKernel\EventListener\SessionListener as BaseSessionListener;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
+@trigger_error(sprintf('The %s class is deprecated since version 3.3 and will be removed in 4.0. Use %s instead.', SessionListener::class, BaseSessionListener::class), E_USER_DEPRECATED);
```

Comment:
Service locators are experimental too, and we are already deprecating arguments/methods/properties based on it in several PRs including merged ones (#21553, #21690, a PSR-11 compatible runtime loader added in twig for this one, ...). 
Of course all of these commits would need to be reverted if we decide to don't keep the feature. I understand what you mean here, but how could we better know if the feature should be kept or not? Please let me know about the decision.